swagconnect/github/connectors.py

```diff
@@ -35,8 +40,39 @@ def from_credentials(cls, credentials):
 
     def get_swagger(self, url: str) -> dict:
         repo_name, branch, path = self._parse_url(url)
+        if not self.validate(path):
+            raise ValidationError("File content type must be JSON, YAML or YML")
+
+        repo = self.client.get_repo(repo_name)
+        contents = repo.get_contents(path).decoded_content.decode()
+        if path.endswith('json'):
+            result = json.loads(contents)
+        else:
+            result = yaml.safe_load(contents)
+        return result
 
-    def _parse_url(self, url: str) -> str:
-        """Parse the given url and return repository name, branch and path to the file or direcotry"""
+    def _parse_url(self, url: str) -> tuple:
+        """
+        Parse the given url and return repository name, branch and path to the file or directory
+        :param url:
+        :return: tuple
+        """
 
         # Return repo name, branch name, path to file
+        uri = urlparse(url)
+        urls = uri.path
+        repo_name, path = urls.split('blob')
+        repo_name, branch = repo_name.strip('/'), path.split('/')[1]
+        path = path.replace('/' + branch + '/', '')
+        return repo_name, branch, path
+
+    def validate(self, path: str) -> bool:
+        """
+        Validate path to YAML or JSON
+        :param url:
+        :return: bool:
+        """
+        path = path.lower()
+        if path.endswith('json') or path.endswith('yml') or path.endswith('yaml'):
```

Comment:
you can just return `return path.endswith('json') or path.endswith('yml') or path.endswith('yaml')`.